### ISSUE TITLE
GenerateToken throws error re: Newtonsoft.Json

### ISSUE THREAD

**Issue by bslatner** [twilio/twilio-csharp#190]
.NET 4.5
Twilio 4.2.1
Twilio.Client 3.4.4.

I get the following message when calling GenerateToken:

```
Could not load file or assembly 'Newtonsoft.Json.Net35, Version=4.0.2.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' or one of its dependencies. The system cannot find the file specified.

Description: An unhandled exception occurred during the execution of the current web request. Please review the stack trace for more information about the error and where it originated in the code.

Exception Details: System.IO.FileNotFoundException: Could not load file or assembly 'Newtonsoft.Json.Net35, Version=4.0.2.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed' or one of its dependencies. The system cannot find the file specified.
```

I require a newer version of Newtonsoft.Json. Binding redirects don't work.

**Comment by carlosdp** [comment]
Hi @bslatner , could you post the stack trace?

**Comment by devinrader** [comment]
Hey @bslatner can you also let us know which version of Visual Studio you are using?

**Comment by bslatner** [comment]
Sure. Using Visual Studio 2015.

```
   at JWT.JsonWebToken.Encode(Object payload, String key, JwtHashAlgorithm algorithm)
   at Twilio.TwilioCapability.GenerateToken(Int32 ttlSeconds) in C:\Users\demo\Documents\GitHub\twilio-csharp\src\Twilio.Client.Capability\TwilioCapability.cs:line 98
   at Twilio.TwilioCapability.GenerateToken() in C:\Users\demo\Documents\GitHub\twilio-csharp\src\Twilio.Client.Capability\TwilioCapability.cs:line 80
   at NoticeSystem.Messaging.Twilio.CapabilityTokenGenerator.GenerateToken(String identity, String deviceId) in C:\Development\NoticeSystem\NoticeSystem.Messaging.Twilio\CapabilityTokenGenerator.cs:line 34
   at NoticeSystem.Messaging.Twilio.CapabilityTokenGenerator.Test() in C:\Development\NoticeSystem\NoticeSystem.Messaging.Twilio\CapabilityTokenGenerator.cs:line 11
```
And here is the whole of the source code that caused the issue, if that helps:

```csharp
    public static class CapabilityTokenGenerator
    {

        private static void Test()
        {
            var token = GenerateToken("bslatner", Guid.NewGuid().ToString("N"));
            System.Diagnostics.Debug.WriteLine(token);
        }

        public static string GetEndpointId(string identity, string deviceId)
        {
            return $"{Properties.Settings.Default.TwilioIpMessagingAppId}:{deviceId}:{identity}";
        }

        public static string GenerateToken(string identity, string deviceId)
        {
            var endpointId = GetEndpointId(identity, deviceId);
            var capability = new TwilioCapability(
                Properties.Settings.Default.TwilioIpMessagingAccountSid,
                Properties.Settings.Default.TwilioIpMessagingAccountAuthorizationToken);
            capability.AllowClientOutgoing(Properties.Settings.Default.TwilioIpMessagingApplicationSid,
                new
                {
                    service_sid = Properties.Settings.Default.TwilioIpMessagingServiceSid,
                    endpoint_id = endpointId,
                    identity
                });

            return capability.GenerateToken();
        }
    }
```

**Comment by devinrader** [comment]
Hey @bslatner.  I was able to replicate the issue in a new console application running .NET 4.6.  

tl;dr - Upgrade the JWT package in Nuget and you should be able to just remove the JSON.NET package all together since none of the Twilio packages depend on it anymore.

The longer version is that the problem appears to be exposed by a change in Nuget 3's behavior.  IIRC, older versions Nuget would default to installing the latest version of a packages declared dependencies, if that package did not explicitly declare a dependency version rule (which the Twilio.Client package does not).

Newer versions of Nuget now default to installing the lowest package dependency version available.  Twilio Client depends on a packages called JWT, which Nuget installs something like version 1.1.0.  That old version of JWT has a dependency on JSON.NET and also did not specify a version rule, so Nuget installed a really old version of JSON.NET.

@carlosdp we should go make sure we have better dependency version rules set on the depdencies defined in our nuspecs.

**Comment by bslatner** [comment]
That did the trick. Thanks!

**Comment by waynebrantley** [comment]
You still have this issue and need to update your package references for nuget.

**Comment by jamsoft** [comment]
Yes, I've just encountered this issue as well.  This ticket should not be closed, it needs addressing.

**Comment by human101101** [comment]
I also ran into this. Please update the package references. Also folks from Twilio need to update your tutorials because none of what is going on here is mentioned.

**Comment by pointandclickinterface** [comment]
Updating the JWT Nuget package in Visual Studio's Nuget Package Manager like @devinrader and @human101101 said worked for me as well.
